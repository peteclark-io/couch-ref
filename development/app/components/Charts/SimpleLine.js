'use strict'

import React from 'react';
import ReactFauxDOM from 'react-faux-dom';

import _ from 'lodash';

import {scaleLinear} from "d3-scale";
import {select} from "d3-selection";
import {max} from "d3-array";

import styles from './SimpleLine.css';

const SimpleLine = React.createClass({

   propTypes: {
     data: React.PropTypes.arrayOf(React.PropTypes.shape({
       title: React.PropTypes.string,
       value: React.PropTypes.number
     })),
     width: React.PropTypes.number
   },

   render: function() {
      if (max(this.props.data) === 0){
        return null;
      }

      var chart = ReactFauxDOM.createElement('svg')
      var xAxis = scaleLinear().domain([0, max(this.props.data.map(i => i.value))]).range([0, 1000]);

      var bars = select(chart)
        .attr('width', '100%')
        .attr('height', '24px')
        .selectAll('g')
        .data(this.props.data)
        .enter()
        .append('g')
        .attr('transform', (d, i) => {
          var trans = (i * 12) + 8;
          return 'translate(0,' + trans +  ')';
        });

      bars.append('text')
          .attr('dy', -1)
          .text((d) => {
            return d.title;
          });

      bars.append('rect')
         .attr('width', (d) => { return xAxis(d.value); })
         .attr('height', 3)
         .attr('fill', (d, i) => { return i % 2 === 0 ? 'rgba(60,90,150,1)' : '#51A8B5'});

      return chart.toReact();
   }
});

export default SimpleLine;
