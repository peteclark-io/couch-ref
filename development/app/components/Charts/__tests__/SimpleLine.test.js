'use strict';

import React from 'react';
import SimpleLine from '../SimpleLine';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';

it('Is null when it has no data.', () => {
   const rendered = mount(
      <SimpleLine />
   );
   expect(rendered.html()).toBeNull();
});

it('Is null when data is 0.', () => {
   const data = [{title: 'Yes', value: 0}, {title: 'No', value: 0}]
   const rendered = mount(
      <SimpleLine data={data} />
   );
   expect(rendered.html()).toBeNull();
});

it('Writes a graph when it has data.', () => {
   const data = [{title: 'Yes', value: 50}, {title: 'No', value: 10}]
   const rendered = mount(
      <SimpleLine data={data} />
   );
   expect(rendered.html()).toMatchSnapshot();
});
