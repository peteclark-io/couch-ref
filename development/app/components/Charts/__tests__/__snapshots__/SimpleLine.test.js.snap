exports[`test Writes a graph when it has data. 1`] = `"<svg width=\"100%\" height=\"24px\"><g transform=\"translate(0,8)\"><text dy=\"-1\" style=\"font-size: 10px;\">Yes</text><rect height=\"3\" fill=\"#51A8B5\" width=\"NaN\"></rect></g><g transform=\"translate(0,20)\"><text dy=\"-1\" style=\"font-size: 10px;\">No</text><rect height=\"3\" fill=\"rgba(60,90,150,1)\" width=\"NaN\"></rect></g></svg>"`;
